framework/src/wifi_manager/wifi_manager.c

```diff
@@ -52,94 +217,98 @@ void __tizenrt_manual_linkset(const char *msg)
-			return WIFI_MANAGER_DEINITIALIZED;	\
-		}	\
-	} while (0)
 
-static wifi_utils_result_e start_dhcp_client(void)
+wifi_manager_result_e
+_convert_scan_info(wifi_manager_scan_info_s **wm_scan_list, wifi_utils_scan_list_s *wu_scan_list)
 {
-	struct dhcpc_state state;
-	int ret;
-
-	g_dhcp_handle = dhcpc_open(CTRL_IFNAME);
-	if (g_dhcp_handle == NULL) {
-		ndbg("Invalid dhcp handle\n");
-		return WIFI_UTILS_FAIL;
-	}
-	ret = dhcpc_request(g_dhcp_handle, &state);
-	if (ret != OK) {
-		dhcpc_close(g_dhcp_handle);
-		g_dhcp_handle = NULL;
-		return WIFI_UTILS_FAIL;
+	wifi_manager_scan_info_s *cur = NULL, *prev = NULL;
+	wifi_utils_scan_list_s *iter = wu_scan_list;
+	while (iter) {
+		cur = (wifi_manager_scan_info_s *)calloc(1, sizeof(wifi_manager_scan_info_s));
+		if (!cur) {
+			_free_scan_info(*wm_scan_list);
+			return WIFI_MANAGER_FAIL;
+		}
+		cur->rssi = iter->ap_info.rssi;
+		cur->channel = iter->ap_info.channel;
+		cur->phy_mode = iter->ap_info.phy_mode;
+		strncpy(cur->ssid, (char *)iter->ap_info.ssid, 32);
+		strncpy(cur->bssid, (char *)iter->ap_info.bssid, 17);
+
+		if (!prev) {
+			*wm_scan_list = cur;
+		} else {
+			prev->next = cur;
+		}
+		prev = cur;
+		iter = iter->next;
 	}
+	return WIFI_MANAGER_SUCCESS;
+}
 
-	netlib_set_ipv4addr(CTRL_IFNAME, &state.ipaddr);
-	netlib_set_ipv4netmask(CTRL_IFNAME, &state.netmask);
-	netlib_set_dripv4addr(CTRL_IFNAME, &state.default_router);
-
-#ifdef CONFIG_ENABLE_IOTIVITY
-	__tizenrt_manual_linkset("gen");
-#endif
-	nvdbg("IP address : %s ----\n", inet_ntoa(state.ipaddr));
 
-	return WIFI_UTILS_SUCCESS;
+void _free_scan_info(wifi_manager_scan_info_s *scan_list)
+{
+	wifi_manager_scan_info_s *iter = scan_list, *prev = NULL;
+	while (iter) {
+		prev = iter;
+		iter = iter->next;
+		free(prev);
+	}
 }
 
-static wifi_utils_result_e stop_dhcp_client(void)
+
+void _convert_state_to_info(connect_status_e *conn, wifi_manager_mode_e *mode, _wifimng_state_e state)
 {
-	if (g_dhcp_handle == NULL) {
-		return WIFI_UTILS_FAIL;
-	} else {
-		dhcpc_close(g_dhcp_handle);
-		g_dhcp_handle = NULL;
+	switch(state) {
+	case WIFIMNG_INIT:
+		*mode = WIFI_NONE;
+		*conn = STATUS_UNKNOWN;
+		break;
+	case WIFIMNG_STA_DISCONNECTED:
+	case WIFIMNG_STA_DISCONNECTING:
+	case WIFIMNG_STA_CONNECTING:
+		*mode = STA_MODE;
+		*conn = AP_DISCONNECTED;
+		break;
+	case WIFIMNG_STA_CONNECTED:
+		*mode = STA_MODE;
+		*conn = AP_CONNECTED;
+		break;
+	case WIFIMNG_SOFTAP:
+		*mode = SOFTAP_MODE;
+		*conn = AP_DISCONNECTED;
+		break;
+	default:
+		// CRITICAL ERROR
+		break;
 	}
-
-	return WIFI_UTILS_SUCCESS;
 }
 
-static wifi_utils_result_e start_dhcp_server(void)
+
+wifi_manager_result_e _start_dhcpd(void)
 {
-	struct in_addr in = {.s_addr = 0x012fa8c0 };
-	netlib_set_ipv4addr(CTRL_IFNAME, &in);
-	in.s_addr = 0x00ffffff;
-	netlib_set_ipv4netmask(CTRL_IFNAME, &in);
-	in.s_addr = 0x012fa8c0;
-	netlib_set_dripv4addr(CTRL_IFNAME, &in);
+	struct in_addr ip = {.s_addr = 0x012fa8c0 };
```

Comment:
hardcode this address? maybe have a Kconfig option for this setting?

Reply:
It would be better